src/VCR/MatchObserver.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace VCR;
+
+class MatchObserver
```

Comment:
Not sure about the `MatchObserver` name. When I read observer I think of the [observer pattern](http://en.wikipedia.org/wiki/Observer_pattern) which isn't really the case here. 

To me it seems like the purpose of that class is to explain mismatches. Maybe `MismatchExplainer`? ;-)